File: frontend/src/services/receiptService.js
import { api } from './api';

export const receiptService = {
  getReceiptById: async (id) => {
    const res = await api.get(`/receipts/${id}`);
    return res.data;
  },

  getReceiptByPaymentId: async (paymentId) => {
    const res = await api.get(`/receipts/payment/${paymentId}`);
    return res.data;
  },

  getReceiptByNumber: async (receiptNumber) => {
    const res = await api.get(`/receipts/number/${encodeURIComponent(receiptNumber)}`);
    return res.data;
  },

  getReceiptsByRollNumber: async (rollNumber) => {
    const res = await api.get(`/receipts/student/${encodeURIComponent(rollNumber)}`);
    return res.data;
  },

  getAllReceipts: async (search) => {
    const query = search ? `?search=${encodeURIComponent(search)}` : '';
    const res = await api.get(`/receipts${query}`);
    return res.data;
  },
};

File: backend/src/main/java/com/university/feems/service/ReceiptService.java
package com.university.feems.service;

import com.university.feems.dto.ReceiptDto;

import java.util.List;

public interface ReceiptService {
    ReceiptDto getReceiptById(Long id);
    ReceiptDto getReceiptByPaymentId(Long paymentId);
    ReceiptDto getReceiptByReceiptNumber(String receiptNumber);
    List<ReceiptDto> getReceiptsByRollNumber(String rollNumber);
    List<ReceiptDto> getAllReceipts();
    List<ReceiptDto> searchReceipts(String query);
}
